logger/logger.go

```diff
@@ -13,47 +13,42 @@
 //  limitations under the License.
 
 // Package logger offers simple logging on GCE.
+// Events are logged to the serial console as well as the event log.
 package logger
 
 import (
 	"fmt"
 	"log"
-	"os"
 	"path/filepath"
 	"runtime"
+	"strings"
 
 	"github.com/tarm/serial"
+	"golang.org/x/sys/windows/svc/eventlog"
 )
 
 var (
-	// Log is the logger's log.Logger.
+	// Log is the serial logger's log.Logger.
 	Log         *log.Logger
+	slInfo      *log.Logger
+	slError     *log.Logger
+	slFatal     *log.Logger
 	initialized bool
 	logger      string
 )
 
-type serialPort struct {
-	Port string
-}
-
-func (s *serialPort) Write(b []byte) (int, error) {
-	c := &serial.Config{Name: s.Port, Baud: 115200}
-	p, err := serial.OpenPort(c)
-	if err != nil {
-		return 0, err
-	}
-	defer p.Close()
-
-	return p.Write(b)
-}
-
 // Init sets up logging and should be called before log functions, usually in
 // the callers main(). Log functions can be called before Init(), but log
 // output will go to COM1.
 func Init(name, port string) {
 	logger = name
 	out := &serialPort{port}
+	// Split logging to the serial port and event log so processes like the
+	// metadata script runnner can log to serial output but not the event log.
 	Log = log.New(out, "", log.Ldate|log.Ltime)
+	if err := slSetup(name); err != nil {
+		log.Fatal(err)
```

Comment:
why not use Log here. the default log goes to stdout and are not easy to see if running as a service.